code/modules/plumbing/plumbers/pill_press.dm

```diff
@@ -1,111 +1,116 @@
+					"ref" = REF(container)
+				)
+				category_item["products"] += list(package_item)
+
+			packaging_types += list(category_item)
+
+	packaging_type = REF(GLOB.reagent_containers[CAT_PILLS][1])
+	_decode_category()
+
 	AddComponent(/datum/component/plumbing/simple_demand, bolt, layer)
 
+/obj/machinery/plumbing/pill_press/examine(mob/user)
+	. = ..()
+	. += span_notice("The [name] currently has [stored_products.len] stored. There needs to be less than [MAX_FLOOR_PRODUCTS] on the floor to continue dispensing.")
+
+/// decode product category from it's type path and returns the decoded typepath
+/obj/machinery/plumbing/pill_press/proc/_decode_category()
+	var/obj/item/reagent_containers/container = locate(packaging_type)
+	if(ispath(container, /obj/item/reagent_containers/pill/patch))
+		packaging_category = CAT_PATCHES
+	else if(ispath(container, /obj/item/reagent_containers/pill))
+		packaging_category = CAT_PILLS
+	else
+		packaging_category = "Bottles"
+	return container
+
 /obj/machinery/plumbing/pill_press/process(seconds_per_tick)
 	if(machine_stat & NOPOWER)
 		return
-	if(reagents.total_volume >= current_volume)
-		if (product == "pill")
-			var/obj/item/reagent_containers/pill/P = new(src)
-			reagents.trans_to(P, current_volume)
-			P.name = trim("[product_name] pill")
-			stored_products += P
-			if(pill_number == RANDOM_PILL_STYLE)
-				P.icon_state = "pill[rand(1,21)]"
-			else
-				P.icon_state = "pill[pill_number]"
-			if(P.icon_state == "pill4") //mirrored from chem masters
-				P.desc = "A tablet or capsule, but not just any, a red one, one taken by the ones not scared of knowledge, freedom, uncertainty and the brutal truths of reality."
-		else if (product == "patch")
-			var/obj/item/reagent_containers/pill/patch/P = new(src)
-			reagents.trans_to(P, current_volume)
-			P.name = trim("[product_name] patch")
-			P.icon_state = patch_style
-			stored_products += P
-		else if (product == "bottle")
-			var/obj/item/reagent_containers/cup/bottle/P = new(src)
-			reagents.trans_to(P, current_volume)
-			P.name = trim("[product_name] bottle")
-			stored_products += P
+
+	//shift & check to account for floating point inaccuracies
+	if(reagents.total_volume + (CHEMICAL_QUANTISATION_LEVEL * 10) >= current_volume)
+		var/obj/item/reagent_containers/container = locate(packaging_type)
+		container = new container(src)
+		var/suffix
+		if(packaging_category == CAT_PILLS)
+			suffix = "Pill"
+		else if(packaging_category == CAT_PATCHES)
+			suffix = "Patch"
+		else
+			suffix = "Bottle"
+		container.name = "[product_name] [suffix]"
+		reagents.trans_to(container, current_volume)
+		stored_products += container
+
+	//dispense stored products on the floor
 	if(stored_products.len)
 		var/pill_amount = 0
 		for(var/thing in loc)
-			if(!istype(thing, /obj/item/reagent_containers/cup/bottle) && !istype(thing, /obj/item/reagent_containers/pill))
+			if(!istype(thing, /obj/item/reagent_containers))
```

Comment:
Github won't let me place a suggestion here, but you should just make the for loop typed: `for(var/obj/item/reagent_containers/thing)`